patches/server/0575-Configurable-door-breaking-difficulty.patch

```diff
@@ -48,15 +64,23 @@ index ddf77350b1762be523eeed760669bc33f634109f..85294ab8eaa654be9358f489cec7e1ae
          }
  
 diff --git a/src/main/java/net/minecraft/world/entity/monster/Zombie.java b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
-index 64222cced10e87438ee3ada26aead0284649be78..e72e9b748b3f3e34baddf01366c703efba50c67c 100644
+index 64222cced10e87438ee3ada26aead0284649be78..026a6d302f7ab729ee8cd3c85f77031fa9ee63bf 100644
 --- a/src/main/java/net/minecraft/world/entity/monster/Zombie.java
 +++ b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
-@@ -97,7 +97,7 @@ public class Zombie extends Monster {
+@@ -97,7 +97,15 @@ public class Zombie extends Monster {
  
      public Zombie(EntityType<? extends Zombie> type, Level world) {
          super(type, world);
 -        this.breakDoorGoal = new BreakDoorGoal(this, Zombie.DOOR_BREAKING_PREDICATE);
-+        this.breakDoorGoal = new BreakDoorGoal(this, com.google.common.base.Predicates.in(world.paperConfig.zombieBreakDoors)); // Paper
++        // Paper start - Adjust breakDoorGoal based in entity
++        if (type.equals(EntityType.ZOMBIFIED_PIGLIN)) {
++            this.breakDoorGoal = new BreakDoorGoal(this, com.google.common.base.Predicates.in(world.paperConfig.zombiePiglingBreakDoors));
++        } else if (type.equals(EntityType.ZOMBIE_VILLAGER)) {
++            this.breakDoorGoal = new BreakDoorGoal(this, com.google.common.base.Predicates.in(world.paperConfig.zombieVillagerBreakDoors));
++        } else {
++            this.breakDoorGoal = new BreakDoorGoal(this, com.google.common.base.Predicates.in(world.paperConfig.zombieBreakDoors));
++        }
++        // Paper end
```

Comment:
A map would also greatly simplify the code here

Reply:
Replaced with a map.. and a default case using Zombie.